lib/log.dart

```diff
@@ -30,3 +32,51 @@ bool debugLog(String message) {
   }());
   return true;
 }
+
+typedef ReportErrorCallback = void Function(String? message, {String? details});
+/// Display an error message in a [SnackBar].
+///
+/// This shows a [SnackBar] containing the message if [ZulipApp] is ready,
+/// otherwise logs it to the console.
+///
+/// If `message` is null, this will clear the existing [SnackBar]s if there
+/// are any.  Useful for promptly dismissing errors.
+///
+/// If `details` is non-null, the [SnackBar] will contain a button that would
+/// open a dialog containing the error details.
+// This gets set in [ZulipApp].  We need this indirection to keep `lib/log.dart`
+// from importing widget code, because the file is a dependency for the rest of
+// the app.
+ReportErrorCallback get reportErrorToUser => (message, {details}) {
```

Comment:
Yeah, that makes sense. I previously created two functions `reportErrorToUserBriefly` (later renamed to `reportErrorToUserInSnackbar`), `reportErrorToUserInDialog`.

In this revision, these two functions were merged into one, losing the suffix, but technically, since it displays a snackbar either way, and only shows a dialog if you click the button on the snackbar, it should be still considered "brief".